sdk/metric/internal/filter.go

```diff
@@ -21,6 +21,24 @@ import (
 	"go.opentelemetry.io/otel/sdk/metric/metricdata"
 )
 
+type filterAgg[N int64 | float64] interface {
+	Aggregator[N]
+
+	// filtered records values for attributes that have been filtered.
+	filtered(N, attribute.Set)
+}
+
+// NewFilter wraps an Aggregator with an attribute filtering function.
+func NewFilter[N int64 | float64](agg Aggregator[N], fn attribute.Filter) Aggregator[N] {
+	if fn == nil {
+		return agg
+	}
+	if fa, ok := agg.(filterAgg[N]); ok {
```

Comment:
why does agg being a filterAgg imply that it is precomputed?  I'm not entirely sure what a `filterAgg` is...

Reply:
Yeah, could probably improve the name here. I'll take a look when I get back.

Reply:
Rename to `precomputedAggregator`. PTAL.